nano/node/nodeconfig.cpp

```diff
@@ -659,6 +661,62 @@ void nano::node_config::deserialize_address (std::string const & entry_a, std::v
 	}
 }
 
+std::string nano::node_config::serialize_database_backend (nano::database_backend backend) const
+{
+	switch (backend)
+	{
+		case nano::database_backend::rocksdb:
+			return "rocksdb";
+		case nano::database_backend::lmdb:
+			return "lmdb";
+	}
+	debug_assert (false);
+}
+
+std::optional<nano::database_backend> nano::node_config::deserialize_database_backend (std::string backend_str) const
+{
+	if (backend_str == "rocksdb")
+		return database_backend::rocksdb;
+	if (backend_str == "lmdb")
+		return database_backend::lmdb;
+	return std::nullopt;
+}
+
+nano::database_backend nano::node_config::get_config_backend (nano::tomlconfig & toml)
+{
+	if (toml.has_key ("database_backend"))
+	{
+		auto backend_str = (toml.get<std::string> ("database_backend"));
+		auto backend = deserialize_database_backend (backend_str);
+		if (backend.has_value ())
+		{
+			return backend.value ();
+		}
+		toml.get_error ().set ("Unknown database_backend type: " + backend_str);
+		return nano::database_backend::lmdb;
+	}
+	// Default to LMDB
+	return nano::database_backend::lmdb;
+}
+
+nano::database_backend nano::node_config::get_default_backend ()
+{
+	auto backend_str = nano::env::get<std::string> ("BACKEND");
```

Comment:
Why don't you name it `DATABASE_BACKEND`?

Reply:
> Why don't you name it `DATABASE_BACKEND`?

It's an existing environment key. I didn't name it

Reply:
NANO_BACKEND would fit better with our convention of prefixing all nano specific env variables with NANO_

Reply:
I have renamed the environment variable to NANO_BACKEND now